code/obj/machinery/clonepod.dm

```diff
@@ -237,17 +237,18 @@ TYPEINFO(/obj/machinery/clonepod)
 
 		if (istype(oldholder))
 			oldholder.clone_generation++
-			src.occupant?.set_mutantrace(oldholder?.mobAppearance?.mutant_race?.type)
 			src.occupant.bioHolder.CopyOther(oldholder, copyActiveEffects = connected?.gen_analysis)
-			if(oldholder?.mobAppearance?.mutant_race?.dna_mutagen_banned)
-				src.occupant?.set_mutantrace(null)
+			src.occupant?.set_mutantrace(oldholder?.mobAppearance?.mutant_race?.type)
+			src.occupant?.set_mutantrace(oldholder?.mobAppearance?.original_mutant_race?.type)
```

Comment:
???

Reply:
`oldHolder` is a copy of the `bioHolder` of the individual that was scanned, which contains an `appearanceHolder`; should the individual in question have been a mutantrace at the time of their clone scan, their `appearanceHolder` will have been modified by the mutantrace so that the indiviual appears as this mutantrace. Their `appearanceHolder` is reset upon the mutantrace being removed, however by nature of cloning, this never occurs, and as a result their appearance holder still takes the form of that mutantrace.

Originally, if their mutantrace was mutagen banned, such as if their mutantrace was a thrall, `set_mutantrace(null)` would have been called, which would result in all their mutantraces being removed, and them being left as a human. This does not occur, as they have no mutantrace to remove in the first place; they only take the appearance of one due to a modified `appearaceHolder`.

To rectify this, their old mutantrace must be reassigned to them, so that it may be removed by the second `set_mutantrace()` call, which has been updated to use `original_mutantrace`, their last non-mutagen-banned mutantrace.

Reply:
uuuugh I hate this, approved 👍